ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/Code.java

```diff
@@ -116,7 +116,7 @@ public boolean isValidJumpSubDestination(
       final EVM evm, final MessageFrame frame, final UInt256 destination) {
     if (!destination.fitsInt()) return false;
     final int jumpDestination = destination.intValue();
-    if (jumpDestination > getSize()) return false;
+    if (jumpDestination >= getSize()) return false;
```

Comment:
I thought this is only an issue for jump sub? I thought we're allowed to have a jumpdest be the last op in the code.

Reply:
it is true . but if the destination is equal to the size of the code it is because we are outside the code (PC == 0). I added a test to validate the fact that we can have a jumpdest at the end of the code

Reply:
Ah! Ok, that makes sense.

Reply:
Wait, shouldn't it be something like `>= getSize() -1`? Because jump sub can't be the last instruction either? So it's technically inbounds but not actually legal.

Reply:
Jumbsub can be the last instruction. there is a sample in the EIP (https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2315.md#subroutine-at-end-of-code) . You can see the Jumpsub opcode at the end of the code. `0x6005565c5d5b60035e` (0x5e)

Reply:
Ok, all I needed to know, LGTM!